src/cls/IPM/ResourceProcessor/WebApplication.cls

```diff
@@ -88,32 +88,46 @@ Method OnBeforePhase(pPhase As %String, ByRef pParams) As %Status
         } ElseIf ((pPhase = "Unconfigure") || (pPhase = "Clean")) {
             // Remove web application
             Do ..DeleteWebApp(tVerbose)
+            Do ..DeleteCodeReadRole($Get(..SecurityProperties("NameSpace")), tVerbose)
         }
     } Catch e {
         Set sc = e.AsStatus()
     }
     Quit sc
 }
 
+Method ComputeAutheNumber(pAuth As %String) As %String
```

Comment:
This shouldn't be necessary with the #{} syntax, but I think it is because of a bug in %IPM.Storage.Module that prevents nesting of system expressions within an ObjectScript expression:

```
	Set tAttrValue = ##class(%IPM.Storage.ModuleSetting.Default).EvaluateAttribute(tAttrValue,.customParams)
	Set attrValue = ##class(%IPM.Utils.Module).%EvaluateSystemExpression(tAttrValue)
	
	// {name} expression does not contain dollar sign; handle separately
	Set name = ..Name
	Set attrValue = ##class(%Regex.Matcher).%New("(?i)\{name\}", attrValue).ReplaceAll($Replace(name,"\","\\"))

	Set regex = ##class(%Regex.Matcher).%New("#\{([^}]+)\}", tAttrValue)
	While regex.Locate() {
		Set expr = regex.Group(1)
		Set value = ..%EvaluateExpression(expr)
		Set $Extract(attrValue, regex.Start, regex.End - 1) = value
		Set regex.Text = attrValue
	}
```

It should be instead:
```
Set regex = ##class(%Regex.Matcher).%New("#\{([^}]+)\}", attrValue)
```

Then I think this method would be unneeded.

Reply:
(this is valid to fix in your PR)